src/openrct2/ride/Vehicle.cpp

```diff
@@ -7275,11 +7275,11 @@ void Vehicle::UpdateSpinningCar()
  */
 void Vehicle::UpdateAdditionalAnimation()
 {
-    uint8_t al{};
-    uint8_t ah{};
+    uint8_t targetFrame{};
+    uint8_t curFrame{};
     uint32_t eax{};
 
-    uint32_t* curVar_C8 = reinterpret_cast<uint32_t*>(&var_C8);
+    uint32_t* curAnimationState = reinterpret_cast<uint32_t*>(&animationState);
```

Comment:
This is horrible, its a cast from 16 bit to 32 bit. I know its been like this from before but this is surely wrong.

Reply:
Would need to introduce a union ideally

Reply:
I think this needs some further looking into, its probably one field and not two.

Reply:
I'm pretty sure it's just one more union situation. There were lots of them originally and it seems this one was missed.

Reply:
Some animation types have different shifts, I think making this 32 bit and then have the right masks is better than the union here.

Reply:
Let’s do this in another PR.